src/osmpbf.rs

```diff
@@ -225,41 +196,27 @@ fn blob_type_from_blob_info(
     })
 }
 
-pub fn build_block_index<P: AsRef<Path>>(path: P) -> io::Result<Vec<BlockIndex>> {
-    let iter = Mutex::new(BlockIndexIterator::new(path)?);
-    let result = Mutex::new(Vec::new());
-    let num_tasks = rayon::current_num_threads();
-    info!(
-        "Building block index with {} parallel task(s)...",
-        num_tasks
-    );
-    rayon::scope(|s| {
-        for _ in 1..num_tasks {
-            s.spawn(|_| {
-                let mut local_result = Vec::new();
-                let mut blob_buf = Vec::new();
-                loop {
-                    let blob = iter.lock().unwrap().next();
-                    let block = match blob {
-                        Some(Ok(BlobInfo::Unknown(start, len, blob))) => {
-                            blob_type_from_blob_info(&mut blob_buf, start, len, blob)
-                        }
-                        Some(Ok(BlobInfo::Header(b))) => Ok(b),
-                        Some(Err(e)) => Err(e),
-                        None => break,
-                    };
-                    match block {
-                        Ok(b) => local_result.push(b),
-                        Err(e) => eprintln!("Skipping block due to error: {}", e),
-                    }
+pub fn build_block_index(pbf_data: &[u8]) -> Vec<BlockIndex> {
+    let mut result: Vec<BlockIndex> = BlockIndexIterator::new(pbf_data)
+        .par_bridge()
+        .filter_map(|blob| {
+            let block = match blob {
+                Ok(BlobInfo::Header(b)) => Ok(b),
+                Ok(BlobInfo::Unknown(start, len, blob)) => {
+                    blob_type_from_blob_info(start, len, blob)
                 }
-                result.lock().unwrap().extend(local_result);
-            });
-        }
-    });
-
-    let mut result = result.into_inner().unwrap();
+                Err(e) => Err(e),
+            };
+            match block {
+                Ok(b) => Some(b),
+                Err(e) => {
+                    eprintln!("Skipping block due to error: {}", e);
+                    None
+                }
+            }
+        })
+        .collect();
```

Comment:
Did you measure how this performs versus the previous version?

Reply:
Yes, no difference I could observe.